scripts/query_rag.py

```diff
@@ -43,8 +42,17 @@
         index_id=args.product_index,
     )
     if args.node is not None:
-        print(repr(storage_context.docstore.get_node(args.node)))
+        print(storage_context.docstore.get_node(args.node))
     else:
         retriever = vector_index.as_retriever(similarity_top_k=args.top_k)
-        for n in retriever.retrieve(args.query):
-            print(repr(n))
+        nodes = retriever.retrieve(args.query)
+        if len(nodes) == 0:
+            print(f"No nodes retrieved for query: {args.query}")
+            exit(1)
+        if args.threshold > 0.0 and nodes[0].score < args.threshold:
+            print(
+                f"Score {nodes[0].score} of the top retrieved node for query '{args.query}' didn't cross the minimal threshold {args.threshold}."
+            )
+            exit(1)
+        for n in nodes:
+            print(n)
```

Comment:
exit(0)?

Reply:
In this case it's implicit. If the script reaches the end without [SystemExit](https://docs.python.org/3.12/library/exceptions.html#SystemExit) being triggered in some way, the `0` exit code will be emitted.

For example: 
```
$ python -c "print('something..something')"; echo $?
something..something
0
```

Reply:
Thanks for the explanation Jiri, marking it as resolved